### PR TITLE
remove dev tag

### PR DESCRIPTION
**Pull request by wilkermichael**
Changes proposed in this PR:
- Remove dev tag so that we don't race against 1.3.x preview image
-

How I've tested this PR:

How I expect reviewers to test this PR:

Checklist:
- [ ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

**Review by curtbushko — APPROVED** [review]

File: charts/consul/Chart.yaml
Comment (DanStough) [inline charts/consul/Chart.yaml:26]: As a complete aside and non-blocking, I wonder if this license should be considering it includes Consul 🤔

**Review by DanStough — APPROVED** [review]

**Comment by DanStough** [comment]
I need to cut the branches in Consul before we merge this.

File: charts/consul/Chart.yaml
Comment (DanStough) [reply charts/consul/Chart.yaml:26]: We discussed this internally and decided to keep it MPL for now.

**Comment by DanStough** [comment]
Waiting for tag changes to merge before this is fixed: https://github.com/hashicorp/consul/pull/19278

**Comment by DanStough** [comment]
FYI @david-yu @NicoletaPopoviciu . This PR was part of staging the `.0` branch to ensure we are testing the code-freeze images. I'll merge once the tests go green.

**Review by david-yu — APPROVED** [review]

**Pull request merged**
